scripts/test-with-kind.sh

```diff
@@ -36,105 +43,135 @@ check_is_installed kubectl "You can install kubectl with the helper scripts/inst
-       APPMESH_PREVIEW=y AWS_ACCOUNT=$AWS_ACCOUNT_ID AWS_REGION=$AWS_REGION make helm-deploy WAIT_PROXY_READY=true
-       check_deployment_rollout appmesh-controller appmesh-system
-       kubectl get pod -n appmesh-system
-        ;;
-      sidecar-v1.22.2.0)
-       APPMESH_PREVIEW=y AWS_ACCOUNT=$AWS_ACCOUNT_ID AWS_REGION=$AWS_REGION make helm-deploy WAIT_PROXY_READY=true SIDECAR_IMAGE_TAG=v1.22.2.0-prod
-       check_deployment_rollout appmesh-controller appmesh-system
-       kubectl get pod -n appmesh-system
-        ;;
-    esac
-
-    echo -n "running integration test type $__type ... "
-    ginkgo -v -r $__type -- --cluster-kubeconfig=${KUBECONFIG} \
-            --cluster-name=${CLUSTER_NAME} --aws-region=${AWS_REGION} \
-            --aws-vpc-id=$__vpc_id
-    echo "ok."
-  done
-}
+  echo running tests
 
-function clean_up {
-    if [ -v "$TMP_DIR" ]; then
-        "${SCRIPTS_DIR}"/delete-kind-cluster.sh -c "$TMP_DIR" || :
-    fi
-    return
-}
+  kubectl delete ns appmesh-system || :
+  kubectl create ns appmesh-system
 
-trap "clean_up" EXIT
+  for __test in ${TEST_SUITES[@]}; do
+    local __test_dir="test/integration/$__test"
+    local __envoy_version="$ENVOY_LATEST_TAG"
 
-aws_check_credentials
+    # /test_app contains test app images, and is not itself an integration test
+    if [[ "$__test" == "test_app" ]]; then
+      continue
+    fi
 
-if [ -z "$AWS_ACCOUNT_ID" ]; then
-    AWS_ACCOUNT_ID=$( aws_account_id )
-fi
+    # This test specifically tests behavior for an older version of envoy
+    if [[ "$__test" == "sidecar-v1.22" ]]; then
+        __envoy_version="$ENVOY_1_22_TAG"
+    fi
 
-ecr_login $AWS_REGION $ECR_URL
+    # Try to delete the controller. Sometimes, carrying over controllers
+    # between tests can cause some minor issues.
+    helm delete -n=appmesh-system appmesh-controller || :
+
+    helm upgrade -i appmesh-controller config/helm/appmesh-controller --namespace appmesh-system \
+      --set image.repository="$CONTROLLER_IMAGE" \
+      --set image.tag="$CONTROLLER_TAG" \
+      --set sidecar.image.repository="$ENVOY_IMAGE" \
+      --set sidecar.image.tag="$__envoy_version" \
+      --set enableBackendGroups=true \
+      --set sidecar.waitUntilProxyReady=true \
+      --set region="$AWS_REGION" \
+      --set accountId="$AWS_ACCOUNT_ID" \
+      --set "env.AWS_DEFAULT_REGION='$AWS_REGION'" \
+      --set "env.AWS_ACCESS_KEY_ID='$AWS_ACCESS_KEY_ID'" \
+      --set "env.AWS_SECRET_ACCESS_KEY='$AWS_SECRET_ACCESS_KEY'" \
+      --set "env.AWS_SESSION_TOKEN='$AWS_SESSION_TOKEN'" \
+      --set sidecar.envoyAwsAccessKeyId="$AWS_ACCESS_KEY_ID" \
+      --set sidecar.envoyAwsSecretAccessKey="$AWS_SECRET_ACCESS_KEY" \
+      --set sidecar.envoyAwsSessionToken="$AWS_SESSION_TOKEN"
+
+    check_deployment_rollout appmesh-controller appmesh-system
+    kubectl get pod -n appmesh-system
+
+    echo -n "running integration test type $__test ... "
+    ginkgo --flakeAttempts=2 -v -r $__test_dir -- \
```

Comment:
I never like doing this, but I added an auto-retry to tests here w/ the `flakeAttempts` flag. At present, the test suite is just not super robust, and this increases the success rate considerably.